building_sim_plugins/building_sim_common/src/door_common.cpp

```diff
@@ -0,0 +1,199 @@
+#include <memory>
+
+#include <building_sim_common/utils.hpp>
+#include <building_sim_common/door_common.hpp>
+
+using namespace std::chrono_literals;
+
+namespace building_sim_common {
+
+rclcpp::Logger DoorCommon::logger() const
+{
+  return rclcpp::get_logger("door_" + _state.door_name);
+}
+
+DoorMode DoorCommon::requested_mode() const
+{
+  return _request.requested_mode;
+}
+
+std::vector<std::string> DoorCommon::joint_names() const
+{
+  std::vector<std::string> joint_names;
+  for (const auto& door : _doors)
+    joint_names.push_back(door.first);
+  
+  return joint_names;
+}
+
+MotionParams& DoorCommon::params()
+{
+  return _params;
+}
+
+bool DoorCommon::is_initialized() const
+{
+  return _initialized;
+}
+
+void DoorCommon::publish_state(const uint32_t door_value,
+  const rclcpp::Time& time)
+{
+  if (!_initialized)
+    return;
+
+  _state.current_mode.value = door_value;
+  _state.door_time = time;
+  _door_state_pub->publish(_state);
+}
+
+
+DoorCommon::DoorCommon(const std::string& door_name,
+  rclcpp::Node::SharedPtr node,
+  const MotionParams& params,
+  const std::string& left_door_joint_name,
+  const std::string& right_door_joint_name,
+  const std::array<double, 2>& left_joint_limits,
+  const std::array<double, 2>& right_joint_limits)
+: _ros_node(std::move(node)),
+  _params(params)
+{
+  if (!left_door_joint_name.empty() && left_door_joint_name != "empty_joint")
+    _doors.insert(std::make_pair(left_door_joint_name,
+      std::make_shared<DoorCommon::DoorElement>(
```

Comment:
Another "why pointer" feedback, can it just be a DoorElement? I also personally like to create a pair with the curly braces (i.e. `_doors.insert({name, element})`) since it's less verbose but that's just a preference

Reply:
Sure, I've corrected it.